### PR TITLE
[Release] 4.2.0

### PR DESCRIPTION
**Pull request by felipeelia**
This PR contains all the changes outlined in the release process (https://github.com/10up/ElasticPress/issues/2772)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,21 +4,49 @@ All notable changes to this project will be documented in this file, per [the Ke
 
 ## [Unreleased]
 
+## [4.2.0] - 2022-05-XX
+ElasticPress 4.2.0 fixes some bugs and introduces some new and exciting changes.
+
+The sync functionality had its JavaScript refactored. Timeouts, memory limits, and general errors are now properly handled and do not make the sync get stuck when performed via the WP-CLI `index` command. There is also a new `get-last-sync` WP-CLI command to check the errors and numbers from the last sync.
+
+Sites with WooCommerce and Protected Content features enabled will have their product lists in the admin leveraging ElasticPress more. Previously, WooCommerce was processing that screen with two different queries, and EP was used only in one of them. Now, it will be only one query, fully handled by ElasticPress. Users wanting to keep the previous behavior can do so by adding `add_filter( 'ep_woocommerce_integrate_admin_products_list', '__return_false' );` to their website's codebase.
```
Comment (brandwaffle) [inline CHANGELOG.md]: "Sites with WooCommerce and Protected Content features enabled will have their product lists in the admin leveraging ElasticPress more." => "We've improved the admin search experience for sites using both WooCommerce and Protected Content."

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,21 +4,49 @@ All notable changes to this project will be documented in this file, per [the Ke
...
+
+Facets are now available through a WordPress block. If you are using the Full Site Editing feature, you can now add ElasticPress Facets to your theme with just a few clicks!
```
Comment (brandwaffle) [inline CHANGELOG.md]: ...just a few clicks! This block has been introduced with a simplified user interface to enable compatibility with Full Site Editing and will continue to be iterated and improved in future versions of the plugin.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,21 +4,49 @@ All notable changes to this project will be documented in this file, per [the Ke
...
 
 ### Changed
 - Facets widgets rendered by a separate class. Props [@felipeelia](https://github.com/felipeelia) via [#2712](https://github.com/10up/ElasticPress/pull/2712).
 - Deprecated `ElasticPress\Feature\Facets\Widget::get_facet_term_html()` in favor of `ElasticPress\Feature\Facets\Renderer::get_facet_term_html()`. Props [@felipeelia](https://github.com/felipeelia) via [#2712](https://github.com/10up/ElasticPress/pull/2712).
+- Log errors and remove indexing status on failed syncs. Props [@felipeelia](https://github.com/felipeelia) via [#2748](https://github.com/10up/ElasticPress/pull/2748).
+- Refactored Sync page JavaScript. Props [@JakePT](https://github.com/JakePT) via [#2738](https://github.com/10up/ElasticPress/pull/2738).
+- Updated admin scripts to use WordPress's version of React instead of including their own. Props [@JakePT](https://github.com/JakePT) via [#2756](https://github.com/10up/ElasticPress/pull/2756).
```
Comment (brandwaffle) [inline CHANGELOG.md]: "including our own"

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,21 +4,49 @@ All notable changes to this project will be documented in this file, per [the Ke
...
 - Full indices removal after e2e tests. Props [@felipeelia](https://github.com/felipeelia) and [@dustinrue](https://github.com/dustinrue) via [#2710](https://github.com/10up/ElasticPress/pull/2710).
 - Usage of the `$return` parameter in `Feature\RelatedPosts::find_related()`. Props [@felipeelia](https://github.com/felipeelia) and [@altendorfme](https://github.com/altendorfme) via [#2719](https://github.com/10up/ElasticPress/pull/2719).
 - Link to API Functions under the Related Posts feature -> Learn more. Props [@felipeelia](https://github.com/felipeelia) and [@burhandodhy](https://github.com/burhandodhy) via [#2729](https://github.com/10up/ElasticPress/pull/2729).
 - Sync of WooCommerce Orders fields when WooCommerce and Protected Content features are enabled. Props [@felipeelia](https://github.com/felipeelia) and [@ecaron](https://github.com/ecaron) via [#2728](https://github.com/10up/ElasticPress/pull/2728).
+- An issue where selecting no features during install would just cause the install page to reload without any feedback. Props [@JakePT](https://github.com/JakePT) via [#2734](https://github.com/10up/ElasticPress/pull/2734).
```
Comment (brandwaffle) [inline CHANGELOG.md]: add props to @tlovett1 for reporting this one.

File: CHANGELOG.md (hunk 3 above)
Comment (felipeelia) [reply CHANGELOG.md]: "their own" in this case would be React's. Probably we should simply make this "Updated admin scripts to use WordPress's version of React"

**Pull request merged**